hyperbolic-rs/rudders/datasets/synopsis.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 The Rudders Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""File with movie lens dataset specific functions"""
+
+import tensorflow as tf
+from pathlib import Path
+import json
+
+def synopsis_to_dict(dataset_path):
+    """
+    Maps raw dataset file to a Dictonary.
+    :param dataset_path: Path to file containing interactions in a format
+        uid::iid::rate::time.
+    :return: Dictionary containing users as keys, and a numpy array of items the user
+      interacted with, sorted by the time of interaction.
+    """
+    filename = "ratings_with_imdb_id_no_gzip.jsonl"
+    samples = {}
+    with tf.io.gfile.GFile(str(dataset_path / filename), 'r') as lines:
+        for line in lines:
+            input = json.loads(line)
+            uid = input['user_id']
+            timestamp = input['timestamp']
+            iid = input['imdb_id']
+            if iid:
+                if uid in samples:
+                    samples[uid].append((iid, timestamp))
+                else:
+                    samples[uid] = [(iid, timestamp)]
+    sorted_samples = {}
+    for uid in samples:
+        sorted_items = sorted(samples[uid], key=lambda p: p[1])
+        sorted_samples[uid] = [pair[0] for pair in sorted_items]
+    num_ratings = len(sorted_samples)
+    return sorted_samples
+
+
+def build_movieid2title(dataset_path):
+    """Builds a mapping between item ids and the title of each item."""
```

Comment:
Fix comment, and maybe name. It's not just ratings here, right? It's a file with synposis?

Reply:
Im not sure what you mean, this is doing the same thing as in the movielens script creating the mapping between ids and movie title. I'll add that the file contains the synopsis as an fyi